sdk/samples/iot/paho_iot_hub_pnp_component_sample.c

```diff
@@ -177,6 +177,145 @@ static void handle_command_message(
     MQTTClient_message* message,
     az_iot_hub_client_method_request* command_request);
 
+/*
+ * This sample extends the IoT Hub Plug and Play Sample above to mimic a Temperature Controller.
+ * This sample connects the IoT Plug and Play enabled device (the Temerature Controller) with the
+ * Digital Twin Model ID (DTMI). If a timeout occurs while waiting for a message from the Azure IoT
+ * Explorer, the sample will continue. If TIMEOUT_MQTT_RECEIVE_MAX_COUNT timeouts occur
+ * consecutively, the sample will disconnect. X509 self-certification is used.
+ *
+ * This Temperature Controller is made up of the following sub-components:
+ * - Device Info
+ * - Temperature Sensor 1
+ * - Temperature Sensor 2
```

Comment:
What's the rationale for duplicating a lot of the markdown doc content into a comment in the sample  source?

Reply:
Because I'm told a lot of people don't look at the readme.  This way the sample source file has the full explanation if someone is only looking at source code.  Unfortunately, the pnp sample explanations are the longest. All the others are pretty succinct.  If there are sections of the pnp explanation that we think are worth removing in the source code to shrink the size, that is totally doable.

A drawback is keeping the readme and the doc in the sample consistent, but I don't expect us to be changing the descriptions once GA happens.  The exception to this is pnp.

Reply:
> A drawback is keeping the readme and the doc in the sample consistent

Right, that was my concern.

> but I don't expect us to be changing the descriptions once GA happens. The exception to this is pnp.

Sounds good.